tests/pypi/parse_requirements/parse_requirements_tests.bzl

```diff
@@ -563,6 +567,62 @@ def _test_different_package_version(env):
 
 _tests.append(_test_different_package_version)
 
+def _test_optional_hash(env):
+    got = parse_requirements(
+        ctx = _mock_ctx(),
+        requirements_by_platform = {
+            "requirements_optional_hash": ["linux_x86_64"],
+        },
```

Comment:
Could you add a test where the `get_index_urls` is set so that we are testing the newly added code?